std/typecons.d

```diff
@@ -143,25 +146,39 @@ public:
         u._p = null;
     }
 
-    ~this()
+    // whether a RefT instance can be @safely destroyed at end of scope
+    static if (__traits(compiles, (RefT t) @safe {})) @trusted ~this()
```

Comment:
Please put this inside the method as such implementation details shouldn't affect the user and as ddoc doesn't know about `static if`